tests/test_cellpylib.py

```diff
@@ -179,6 +189,20 @@ def test_table_walk_through_decreasing_isotropic(self):
                                                    isotropic=True)
         self.assertEqual(new_lambda, 0.0)
 
+    def test_table_walk_through_lambda_is_actual_lambda(self):
+        table = {'101': 1, '111': 0, '011': 0, '110': 1, '000': 0, '100': 0, '010': 0, '001': 1}
+        table, new_lambda = cpl.table_walk_through(table, lambda_val=0.8148148148148148, k=3, r=1, quiescent_state=0,
+                                                   isotropic=True)
+        self.assertEqual(new_lambda, 0.8148148148148148)
+
+    def test_table_rule(self):
+        r = cpl.table_rule([1, 1, 0], {'101': 1, '111': 0, '011': 0, '110': 1, '000': 0, '100': 0, '010': 0, '001': 1})
+        self.assertEqual(r, 1)
+
+        with pytest.raises(Exception) as e:
+            cpl.table_rule([0, 0, 1], {'101': 1, '111': 0, '011': 0, '110': 1, '000': 0, '100': 0, '010': 0})
+        self.assertTrue("could not find state '001' in table" in str(e))
```

Comment:
As a comment, I want to make you aware of the following pytest feature for matching error messages:

```python
with pytest.raises(Exception, match="could not find state '001'"):
    cpl.table_rule(...)
```

(No need to change the code base for that, just thought I'd bring it up in case you don't know about it.)